packages/app-root/src/components/PageContextProviders.js

```diff
@@ -17,27 +18,64 @@ const GlobalStyle = createGlobalStyle`
 /**
   Context for every page:
   - global page styles.
-  - Zooniverse Grommet theme.
+  - Zooniverse Grommet theme and themeMode.
   - Panoptes auth (user account and admin mode.)
 */
 export default function PageContextProviders({ children }) {
+  const [themeMode, setThemeMode] = useState('auto') // Grommet 'auto' looks for browser dark mode preference like below
+
   const { data: user, error, isLoading } = usePanoptesUser()
   const { adminMode, toggleAdmin } = useAdminMode(user)
   const authContext = { adminMode, error, isLoading, toggleAdmin, user }
 
+  useEffect(() => {
+    const isBrowser = typeof window !== 'undefined'
+    const localStorage = isBrowser ? window.localStorage : null
+
+    // If no theme item in localStorage, see if the user's browser settings prefer dark mode
+    // If theme key is in localStorage, use that for themeMode
+    if (isBrowser && !localStorage?.getItem('theme') ) {
+      const prefersDark = window.matchMedia('(prefers-color-scheme: dark)').matches
+      if (prefersDark) {
+        setThemeMode('dark')
+        localStorage?.setItem('theme', 'dark')
+      }
+    } else {
+      setThemeMode(localStorage?.getItem('theme'))
+    }
+  }, [])
+
```

Comment:
Okay I've got a working theme mode now. If a user's browser prefers dark mode and the `theme` key doesn't exist in localStorage, we'll set `themeMode` to dark mode. Otherwise, grab the stored `theme` key from localStorage and set it as `themeMode`. Side note that Grommet accepts `'auto'` for the theme which says to look for the browser's dark mode preference: https://github.com/grommet/grommet/blob/14cb2cf6be1474d1c8621831fc1d37969846cf4b/src/js/components/Grommet/Grommet.js#L91

I would love to put this useEffect() logic in useState() initialization instead, but when I tried it, I couldn't get the theme UX to work properly. Without useEffect, PageContextProviders is rendered on the server as light mode and even when the `themeMode` state variable was updated via grabbing it from localStorage, components wouldn't re-render with dark mode.

The tradeoff here is that there will be a flash of light mode content when a user has theme = 'dark' in localStorage, but their browser preference is 'light'. If a user's browser preference is 'dark' and the theme is stored as 'dark' in localStorage there is no flash of light mode content. Similarly, if a user's browser preference is set to 'dark', but they have theme = 'light' in localStorage, there's a brief flash of dark mode before re-rendering light mode.